test/e2e/k8sreporter/reporter.go

```diff
@@ -1,175 +1,67 @@
+// SPDX-License-Identifier:Apache-2.0
+
 package k8sreporter
 
 import (
-	"context"
-	"encoding/json"
-	"fmt"
-	"os"
-	"path"
+	"log"
 	"strings"
-	"sync"
+	"time"
 
-	"github.com/kennygrant/sanitize"
-	testclient "github.com/metallb/metallb-operator/test/e2e/client"
-	"github.com/onsi/ginkgo/config"
-	"github.com/onsi/ginkgo/types"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/openshift-kni/k8sreporter"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	metallbv1beta2 "go.universe.tf/metallb/api/v1beta2"
 	corev1 "k8s.io/api/core/v1"
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
-var operatorNameSpace string
-
-type KubernetesReporter struct {
-	sync.Mutex
-	clients    *testclient.ClientSet
-	reportPath string
-}
-
-func New(clients *testclient.ClientSet, nameSpace string, reportPath string) *KubernetesReporter {
-	operatorNameSpace = nameSpace
-	return &KubernetesReporter{clients: clients, reportPath: reportPath}
-}
-
-func (r *KubernetesReporter) SpecSuiteWillBegin(config config.GinkgoConfigType, summary *types.SuiteSummary) {
-
-}
-
-func (r *KubernetesReporter) BeforeSuiteDidRun(setupSummary *types.SetupSummary) {
-	r.Cleanup()
-}
-
-func (r *KubernetesReporter) SpecWillRun(specSummary *types.SpecSummary) {
-}
-
-func (r *KubernetesReporter) SpecDidComplete(specSummary *types.SpecSummary) {
-	r.Lock()
-	defer r.Unlock()
-
-	if !specSummary.HasFailureState() {
-		return
-	}
-	f, err := logFileFor(r.reportPath, "all", "")
-	if err != nil {
-		return
-	}
-	defer f.Close()
-	fmt.Fprintln(f, "Starting dump for failed spec", specSummary.ComponentTexts)
-	dirName := sanitize.BaseName(strings.Join(specSummary.ComponentTexts, ""))
-	dirName = strings.Replace(dirName, "Top-Level", "", 1)
-	r.Dump(dirName)
-	fmt.Fprintln(f, "Finished dump for failed spec")
-}
-
-func (r *KubernetesReporter) Dump(dirName string) {
-	err := os.Mkdir(path.Join(r.reportPath, dirName), 0755)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to create test dir: %v\n", err)
-		return
-	}
-
-	r.logNodes(dirName)
-	r.logPods(operatorNameSpace, dirName)
-	r.logLogs(func(p *corev1.Pod) bool {
-		return !strings.Contains(p.Namespace, "metallb")
-	}, dirName)
-
-}
-
-// Cleanup cleans up the current content of the artifactsDir
-func (r *KubernetesReporter) Cleanup() {
-}
-
-func (r *KubernetesReporter) logPods(namespace string, dirName string) {
-	pods, err := r.clients.Pods(namespace).List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
-		return
-	}
-	for _, pod := range pods.Items {
-		f, err := logFileFor(r.reportPath, dirName, pod.Namespace+"-pods_specs")
+func InitReporter(kubeconfig, path, namespace string) *k8sreporter.KubernetesReporter {
```

Comment:
I'd call this either "Init" or even better "New" because it creates a new instance.
InitReporter duplicates the "reporter" part that is already contained in the package.

Reply:
Done